src/test/lambda/local/detectLocalTemplates.test.ts

```diff
@@ -0,0 +1,295 @@
+                uri: vscode.Uri.file(workspaceFolderPath)
+            } as any as vscode.WorkspaceFolder
+        )
+
+        const templates: vscode.Uri[] = []
+        for await (const template of result) {
+            templates.push(template)
+        }
+
+        assert.strictEqual(templates.length, 2)
+        assert.strictEqual(templates[0].fsPath, normalizePath(workspaceFolderChildPath1, 'template.yaml'))
+        assert.strictEqual(templates[1].fsPath, normalizePath(workspaceFolderChildPath2, 'template.yaml'))
+    })
+
+    it('Detects multiple templates when both template.yml and template.yaml exist in a folder', async () => {
+        const workspaceFolderPath = normalizePath('my', 'workspace', 'folder')
+        const workspaceFolderChildPath = normalizePath(workspaceFolderPath, 'child')
+
+        const result = detectLocalTemplates(
+            {
+                async accessAsync(_path: string | Buffer): Promise<void> {
+                    switch (_path) {
+                        case normalizePath(workspaceFolderChildPath, 'template.yml'):
+                        case normalizePath(workspaceFolderChildPath, 'template.yaml'):
+                            return
+                        default:
+                            throw new Error(`No file found at path: '${_path}'`)
+                    }
+                },
+
+                async readdirAsync(
+                    _path: string | Buffer,
+                    options?: {
+                        encoding: BufferEncoding | null
+                        withFileTypes?: false
+                    } | BufferEncoding | undefined | null
+                ): Promise<string[]> {
+                    switch (_path) {
+                        case workspaceFolderPath:
+                            return [ 'child' ]
+                        case workspaceFolderChildPath:
+                            return [ 'template.yml', 'template.yaml' ]
+                        default:
+                            throw new Error(`Unexpected path: '${_path}'`)
+                    }
+                },
+
+                async statAsync(
+                    _path: string | Buffer
+                ): Promise<filesystem.Stats> {
+                    return {
+                        isDirectory() {
+                            return true
+                        }
+                    } as any as filesystem.Stats
+                }
+            },
+            {
+                uri: vscode.Uri.file(workspaceFolderPath)
+            } as any as vscode.WorkspaceFolder
+        )
+
+        const templates: vscode.Uri[] = []
+        for await (const template of result) {
+            templates.push(template)
+        }
+
+        assert.strictEqual(templates.length, 2)
+        assert.strictEqual(templates[0].fsPath, normalizePath(workspaceFolderChildPath, 'template.yaml'))
+        assert.strictEqual(templates[1].fsPath, normalizePath(workspaceFolderChildPath, 'template.yml'))
```

Comment:
The order here (.yaml, then .yml) is interesting in that above it is reversed. Is the order deterministic?